bookserver/routers/assessment.py

```diff
@@ -39,15 +39,20 @@
 # getAssessResults
 # ----------------
 @router.post("/results")
-async def get_assessment_results(request_data: AssessmentRequest):
-    # if (
-    #     verifyInstructorStatus(auth.user.course_name, auth.user) and request.vars.sid
-    # ):  # retrieving results for grader
-    #     sid = request.vars.sid
-    # else:
-    #     sid = auth.user.username
-
-    # Identify the correct event and query the database so we can load it from the server
+async def get_assessment_results(
+    request_data: AssessmentRequest,
+    request: Request,
+):
+
+    # if the user is not logged in an HTTP 401 will be returned.
+    # Otherwise if the user is an instructor then use the provided
+    # sid (it could be any student in the class) If none is provided then
+    # use the user objects username
+    if await is_instructor(request):
+        if not request_data.sid:
+            request_data.sid = request.state.user.username
+    else:
+        request_data.sid = request.state.user.username
```

Comment:
This works as is, but I wonder if we should be raising a 401 here in the case where the user is not an instructor AND is trying to send along a user id that is not theirs.  

The way it will always return a result for a logged in user, it will just ignore the sid field of the request_data parameter if the user is not an instructor.
If someone really is trying to spoof the system I like the idea of not giving them anything and the 401 says "caught ya"

Reply:
I agree!